### DISCUSSION TITLE
0.25.0

### DISCUSSION THREAD

**Discussion by AmrDeveloper** [AmrDeveloper/GQL#113] [Announcements]
Support JOIN more than two tables togther in same query.
- Handle hidden selection with multi tables.
- Support query `datetime` from diffs table.
- Implement LEFT, RIGHT, INNER and CROSS JOINS operation.
- Implement JOIN predicate using `ON` keyword.
- Simplifiy the DataProvider Design.
- Implement Bitwise `xor` operator.
- Change XOR operator to match postgresql.
- Implement `ARRAY_SHUFFLE` Array function.
- Implement `ARRAY_POSITION` Array Function.
- Implement `ARRAY_DIMS` Array Function.
- Support `Infinity` and `NaN` values.
- Support `OUTER` keyword.

![joins with split](https://github.com/AmrDeveloper/GQL/assets/23631699/cfbf465f-7e3d-47f9-a6ee-1bbf05ae5248)

<hr /><em>This discussion was created from the release <a href='https://github.com/AmrDeveloper/GQL/releases/tag/0.25.0'>0.25.0</a>.</em>

**Comment by amrhossamdev** [comment]
Great work!

**Reply by AmrDeveloper** [reply]
Thank you @amrhossamdev
